File: SpringBootTaskManagerAPI/taskmanager/src/main/java/com/example/taskmanager/entity/TaskUpdate.java
package com.example.taskmanager.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Entity
@ToString
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "TASKUPDATE")
public class TaskUpdate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "TASKUPDATE_ID", nullable = false)
    private Integer taskUpdateId;

    @Column(name = "UPDATE_DATE", nullable = false)
    private LocalDateTime updateDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "TASK_ID", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Task task;

    @Column(name = "COMMENT", nullable = true)
    private String comment;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "STATUS_ID", nullable = false)
    @OnDelete(action = OnDeleteAction.RESTRICT)
    private Status status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "EMPLOYEE_ID", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Employee employee;
}

File: app/src/main/java/com/cst3115/enterprise/taskmanager/model/TaskUpdate.kt
package com.cst3115.enterprise.taskmanager.model

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class TaskUpdate(
    @SerializedName("taskUpdateId") val taskUpdateId: Int,
    @SerializedName("updateDate") val updateDate: String, // or LocalDateTime if you're parsing
    @SerializedName("taskId") val taskId: Int,
    @SerializedName("comment") val comment: String,
    @SerializedName("statusId") val statusId: Int,
    @SerializedName("employeeId") val employeeId: Int
)
